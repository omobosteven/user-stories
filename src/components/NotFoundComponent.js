import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundComponent = () => {
  return (
    <div data-testid="not-found">
      <h1>404!</h1>
      <p>
        Page not Found
        <Link to="/" data-testid="home-link"> Click here</Link>
      </p>
    </div>
  );
};

export default NotFoundComponent;
