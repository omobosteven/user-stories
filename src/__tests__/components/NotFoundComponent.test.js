import React from 'react';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
import {
  render, screen
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import NotFoundComponent from '../../components/NotFoundComponent';

const renderWithRouter = (component) => {
  const history = createMemoryHistory();
  return {
    ...render(
      <Router history={history}>
        {component}
      </Router>
    )
  };
};


describe('NotFoundComponent Test', () => {
  it('should take a snapshot', () => {
    const {
      asFragment, container
    } = renderWithRouter(<NotFoundComponent />);
    const notFoundDiv = screen.getByTestId('not-found');
    const link = screen.getByTestId('home-link');

    expect(asFragment(<NotFoundComponent />)).toMatchSnapshot();
    expect(container.innerHTML).toMatch('Page not Found');
    expect(notFoundDiv).toContainElement(link);
  });
});
