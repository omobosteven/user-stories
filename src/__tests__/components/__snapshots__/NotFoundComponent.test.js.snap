// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundComponent Test should take a snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="not-found"
  >
    <h1>
      404!
    </h1>
    <p>
      Page not Found
      <a
        data-testid="home-link"
        href="/"
      >
         Click here
      </a>
    </p>
  </div>
</DocumentFragment>
`;
